pennylane/labs/resource_estimation/templates/stateprep.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Resource operators for state preparation templates."""
+import math
+
+from pennylane.labs import resource_estimation as plre
+from pennylane.labs.resource_estimation.qubit_manager import AllocWires
+from pennylane.labs.resource_estimation.resource_operator import (
+    CompressedResourceOp,
+    GateCount,
+    ResourceOperator,
+    resource_rep,
+)
+
+
+# pylint: disable=arguments-differ, unused-argument
+class ResourceUniformStatePrep(ResourceOperator):
+    r"""Resource class for preparing a uniform superposition, :
+
+    .. math:
+        \frac{1}{\sqrt{2^{k}L}} \sum_{l=0}^{2^{k}L-1} |l\rangle
+
+    where :math:`L` is odd, starting from the :math:`|0\rangle` state.
+    This operation uses amplitude amplification to prepare a uniform superposition over :math:`l` basis states, where :math:`l` is not a power of 2.
+
+    Args:
+        num_states (int): the number of states over which the uniform superposition is being prepared
+        wires (Sequence[int], optional): the wires the operation acts on
+
+    Resources:
+        The resources are obtained from Figure 12 in `Babbush et al. (2018) <https://arxiv.org/pdf/1805.03662>`_.
+        The circuit uses amplitude amplification to prepare a uniform superposition over :math:`l` basis states, where :math:`l = 2^{k}L`.
+
+    **Example**
+
+    The resources for this operation are computed using:
+
+    >>> unif_state_prep = plre.ResourceUniformStatePrep(10)
+    >>> print(plre.estimate_resources(unif_state_prep))
+    --- Resources: ---
+     Total qubits: 5
+     Total gates : 124
+     Qubit breakdown:
+      clean qubits: 1, dirty qubits: 0, algorithmic qubits: 4
+     Gate breakdown:
+      {'Hadamard': 16, 'X': 12, 'CNOT': 4, 'Toffoli': 4, 'T': 88}
+    """
+
+    resource_keys = {"num_states"}
+
+    def __init__(self, num_states, wires=None):
+        self.num_states = num_states
+        k = (num_states & -num_states).bit_length() - 1
```

Comment:
[optional] typically using full names instead of single letters is preferred. Applies to both k and L.